cmd/keys_rotate.go

```diff
@@ -0,0 +1,360 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var keysRotateCmd = &cobra.Command{
+	Use:   "rotate <offline key archive>",
+	Short: "Rotate root signing key used by the Factory",
+	Run:   doKeyRotation,
+	Args:  cobra.MinimumNArgs(1),
+}
+
+func init() {
+	keysCmd.AddCommand(keysRotateCmd)
+}
+
+type rotationExitCode int
+
+const (
+	ErrUnknown               rotationExitCode = -1
+	RotationSuccess          rotationExitCode = 0
+	ErrUnpackCredentials     rotationExitCode = 1
+	ErrPullCredentials       rotationExitCode = 2
+	ErrParseCredentials      rotationExitCode = 3
+	ErrRegenerateCredentials rotationExitCode = 4
+	ErrAddNewCredentials     rotationExitCode = 5
+	ErrRemoveOldCredentials  rotationExitCode = 6
+	ErrSignCredentials       rotationExitCode = 7
+	ErrPackCredentials       rotationExitCode = 8
+	ErrPushCredentials       rotationExitCode = 9
+)
+
+var rotationErrorExitCodes = [9]rotationExitCode{
+	ErrUnpackCredentials, ErrPullCredentials, ErrParseCredentials, ErrRegenerateCredentials, ErrAddNewCredentials,
+	ErrRemoveOldCredentials, ErrSignCredentials, ErrPackCredentials, ErrPushCredentials}
+
+func doKeyRotation(cmd *cobra.Command, args []string) {
+	factory := viper.GetString("factory")
+	logrus.Debugf("Rotating factory keys for: %s", factory)
+	credentialsPath := args[0]
+	credentialsBackupPath := fmt.Sprintf("%s.bak", credentialsPath)
+	if err := verifyDocker(); err != nil {
+		exitf("invalid environment, %s", err)
+	}
+	const aktualizrImageName = "hub.foundries.io/aktualizr"
+	const rotateCmd = `#!/usr/bin/python3
+import datetime
+import json
+import os
+import subprocess
+import sys
+from tempfile import TemporaryDirectory
+
+def cmd(*args, cwd=None):
+    print('running: %s' % ' '.join(args))
+    p = subprocess.Popen(
+        args, cwd=cwd,stderr=subprocess.STDOUT, stdout=subprocess.PIPE)
+    for line in p.stdout:
+        sys.stdout.buffer.write(line)
+    p.wait()
+    if p.returncode != 0:
+        raise subprocess.CalledProcessError(p.returncode, args)
```

Comment:
Now that you've removed the original formatting I was doing the output, there's no point in all this. calling subprocess.check_call(args, cwd=cwd) would be the same.

Reply:
the python script fails if I use `subprocess.check_call`